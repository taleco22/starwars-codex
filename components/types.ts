export type Character = React.FC<Subject<ICharacters>>
export type Starship = React.FC<Subject<IStarships>>
export type Planet = React.FC<Subject<IPlanets>>
export type Specie = React.FC<Subject<ISpecies>>
export type Film = React.FC<Subject<IFilms>>
export type Vehicle = React.FC<Subject<IVehicles>>

interface Subject<T> {
  results: T[]
}

export interface ICharacters {
  readonly name: string
  readonly mass: string
  readonly height: string
  readonly hair_color: string
  readonly skin_color: string
  readonly eye_color: string
  readonly birth_year: string
  readonly gender: string
  readonly homeworld: string
  readonly films: string[]
  readonly species: string[]
  readonly vehicles: string[]
  readonly starships: string[]
  readonly created: string
  readonly edited: string
  readonly url: string
}

export interface IStarships {
  readonly MGLT: string
  readonly cargo_capacity: string
  readonly consumables: string
  readonly cost_in_credits: string
  readonly created: string
  readonly crew: string
  readonly edited: string
  readonly hyperdrive_rating: string
  readonly length: string
  readonly manufacturer: string
  readonly max_atmosphering_speed: string
  readonly model: string
  readonly name: string
  readonly passengers: string
  readonly films: string[]
  readonly pilots: string[]
  readonly starship_class: string
  readonly url: string
}

export interface IPlanets {
  readonly climate: string
  readonly created: string
  readonly diameter: string
  readonly edited: string
  readonly films: string[]
  readonly gravity: string
  readonly name: string
  readonly orbital_period: string
  readonly population: string
  readonly residents: string[]
  readonly rotation_period: string
  readonly surface_water: string
  readonly terrain: string
  readonly url: string
}

export interface ISpecies {
  readonly average_height: string
  readonly average_lifespan: string
  readonly classification: string
  readonly created: string
  readonly designation: string
  readonly edited: string
  readonly eye_colors: string
  readonly hair_colors: string
  readonly homeworld: string
  readonly language: string
  readonly name: string
  readonly people: string[]
  readonly films: string[]
  readonly skin_colors: string
  readonly url: string
}

export interface IFilms {
  readonly title: string
  readonly episode_id: number
  readonly director: string
  readonly producer: string
  readonly opening_crawl: string
  readonly characters: string[]
  readonly created: string
  readonly edited: string
  readonly planets: string[]
  readonly release_date: string
  readonly species: string[]
  readonly starships: string[]
  readonly url: string
  readonly vehicles: string[]
}

export interface IVehicles {
  readonly cargo_capacity: string
  readonly consumables: string
  readonly cost_in_credits: string
  readonly created: any
  readonly crew: string
  readonly edited: any
  readonly length: string
  readonly manufacturer: string
  readonly max_atmosphering_speed: string
  readonly model: string
  readonly name: string
  readonly passengers: string
  readonly pilots: string[]
  readonly films: string[]
  readonly url: string
  readonly vehicle_class: string
}
