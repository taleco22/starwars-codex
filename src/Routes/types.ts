import { RouteComponentProps } from "react-router-dom";

export interface NavigationProps {
  fn: void;
}

export type CategoriesProps = NavigationProps & RouteComponentProps;

export interface HomeProps extends NavigationProps, RouteComponentProps {
  start: Function;
}

export type SParams = { theme: string };
export type SubjectProps = RouteComponentProps<SParams> & NavigationProps;

export interface SelectComponent {
  [key: string]: JSX.Element;
  people: JSX.Element;
  planets: JSX.Element;
  vehicles: JSX.Element;
  starships: JSX.Element;
  species: JSX.Element;
  films: JSX.Element;
}
