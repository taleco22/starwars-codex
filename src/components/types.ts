interface ModalProps {
  closeModal: React.MouseEventHandler;
}

export type Character = React.FC<Subject<Characters>>;
export type Starship = React.FC<Subject<Starships>>;
export type Planet = React.FC<Subject<Planets>>;
export type Specie = React.FC<Subject<Species>>;
export type Film = React.FC<Subject<Films>>;
export type Vehicle = React.FC<Subject<Vehicles>>;

interface Subject<T> extends ModalProps {
  results: T[];
}

interface Characters {
  readonly name: string;
  readonly mass: string;
  readonly height: string;
  readonly hair_color: string;
  readonly skin_color: string;
  readonly eye_color: string;
  readonly birth_year: string;
  readonly gender: string;
  readonly homeworld: string;
  readonly films: string[];
  readonly species: string[];
  readonly vehicles: string[];
  readonly starships: string[];
  readonly created: string;
  readonly edited: string;
  readonly url: string;
}

interface Starships {
  readonly MGLT: string;
  readonly cargo_capacity: string;
  readonly consumables: string;
  readonly cost_in_credits: string;
  readonly created: string;
  readonly crew: string;
  readonly edited: string;
  readonly hyperdrive_rating: string;
  readonly length: string;
  readonly manufacturer: string;
  readonly max_atmosphering_speed: string;
  readonly model: string;
  readonly name: string;
  readonly passengers: string;
  readonly films: string[];
  readonly pilots: string[];
  readonly starship_class: string;
  readonly url: string;
}

interface Planets {
  readonly climate: string;
  readonly created: string;
  readonly diameter: string;
  readonly edited: string;
  readonly films: string[];
  readonly gravity: string;
  readonly name: string;
  readonly orbital_period: string;
  readonly population: string;
  readonly residents: string[];
  readonly rotation_period: string;
  readonly surface_water: string;
  readonly terrain: string;
  readonly url: string;
}

interface Species {
  readonly average_height: string;
  readonly average_lifespan: string;
  readonly classification: string;
  readonly created: string;
  readonly designation: string;
  readonly edited: string;
  readonly eye_colors: string;
  readonly hair_colors: string;
  readonly homeworld: string;
  readonly language: string;
  readonly name: string;
  readonly people: string[];
  readonly films: string[];
  readonly skin_colors: string;
  readonly url: string;
}

interface Films {
  readonly title: string;
  readonly episode_id: number;
  readonly director: string;
  readonly producer: string;
  readonly opening_crawl: string;
  readonly characters: string[];
  readonly created: string;
  readonly edited: string;
  readonly planets: string[];
  readonly release_date: string;
  readonly species: string[];
  readonly starships: string[];
  readonly url: string;
  readonly vehicles: string[];
}

interface Vehicles {
  readonly cargo_capacity: string;
  readonly consumables: string;
  readonly cost_in_credits: string;
  readonly created: any;
  readonly crew: string;
  readonly edited: any;
  readonly length: string;
  readonly manufacturer: string;
  readonly max_atmosphering_speed: string;
  readonly model: string;
  readonly name: string;
  readonly passengers: string;
  readonly pilots: string[];
  readonly films: string[];
  readonly url: string;
  readonly vehicle_class: string;
}
