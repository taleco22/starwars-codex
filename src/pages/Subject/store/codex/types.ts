/**
 * Action Types
 */

export enum CodexTypes {
  SET_DATA = "data/set_data",
  FAILURE = "data/failure",
  IS_LOADING = "data/is_loading",
  ASYNC_GET_DATA = "data/async_get_data",
}

/**
 * Data Types
 */

export interface Codex {
  count: number
  next: string | null
  previous: string | null
  results: any[]
}

/**
 * State Types
 */

export interface CodexState {
  data: Codex
  isLoading: boolean
}

export interface CodexAction {
  type: string
  payload: CodexState
}
