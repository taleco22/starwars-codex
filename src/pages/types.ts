import { RouteComponentProps } from "react-router-dom"

export type CategoriesProps = RouteComponentProps

export type HomeProps = RouteComponentProps

export type SParams = { theme: string }
export type SubjectProps = RouteComponentProps<SParams>

export interface SelectComponent {
  [key: string]: JSX.Element
  people: JSX.Element
  planets: JSX.Element
  vehicles: JSX.Element
  starships: JSX.Element
  species: JSX.Element
  films: JSX.Element
}

export type URL =
  | "people"
  | "planets"
  | "starships"
  | "vehicles"
  | "films"
  | "species"
