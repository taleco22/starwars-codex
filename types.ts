export type SParams = { theme: string }

export interface SelectComponent {
  [key: string]: JSX.Element
  people: JSX.Element
  planets: JSX.Element
  vehicles: JSX.Element
  starships: JSX.Element
  species: JSX.Element
  films: JSX.Element
}

export type URL =
  | "people"
  | "planets"
  | "starships"
  | "vehicles"
  | "films"
  | "species"
