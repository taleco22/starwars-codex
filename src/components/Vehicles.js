import React from "react";
import * as S from "../styles/styled/index";

const Vehicles = ({results, closeModal}) => {
  return (
    <>
      {results.map((v, i) => (
        <S.Item key={i}>
          <S.Button>
            <h1>{v.name}</h1>
          </S.Button>
          <S.Modal>
            <S.X onClick={closeModal} />
            <h1>{v.name}</h1>
            <p>model: {v.model}</p>
            <p>manufacturer: {v.manufacturer}</p>
            <p>passengers: {v.passengers}</p>
            <p>pilots: {v.pilots}</p>
            <p>const in credits: {v.cost_in_credits}</p>
          </S.Modal>
        </S.Item>
      ))}
    </>
  );
};

export default Vehicles;
