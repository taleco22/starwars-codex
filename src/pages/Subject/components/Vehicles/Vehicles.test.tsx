import React from "react"
import renderer from "react-test-renderer"
import { IVehicles } from "../types"
import Vehicles from "./Vehicles"

describe("Starships.tsx", () => {
  it("render Starships component with correct data", () => {
    const results: IVehicles = {
      cargo_capacity: "50000",
      consumables: "2 months",
      cost_in_credits: "150000",
      created: "2014-12-10T15:36:25.724000Z",
      crew: "46",
      edited: "2014-12-10T15:36:25.724000Z",
      length: "36.8",
      manufacturer: "Corellia Mining Corporation",
      max_atmosphering_speed: "30",
      model: "Digger Crawler",
      name: "Sand Crawler",
      passengers: "30",
      pilots: [],
      films: ["https://swapi.dev/api/films/1/"],
      url: "https://swapi.dev/api/vehicles/4/",
      vehicle_class: "wheeled",
    }
    const component = renderer.create(<Vehicles results={[results]} />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
