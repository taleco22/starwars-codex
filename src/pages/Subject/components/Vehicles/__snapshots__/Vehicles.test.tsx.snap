// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Starships.tsx render Starships component with correct data 1`] = `
<li
  className="sc-AxiKw sc-AxheI bUcjaj"
>
  <button
    className="sc-AxhCb sc-Axmtr gFtPZS"
  >
    <h1>
      Sand Crawler
    </h1>
  </button>
  <div
    className="sc-fzozJi jkmZKH"
  >
    <div
      className="sc-AxmLO ebPzVP"
      onClick={[Function]}
    />
    <h1>
      Sand Crawler
    </h1>
    <p>
      model: 
      Digger Crawler
    </p>
    <p>
      manufacturer: 
      Corellia Mining Corporation
    </p>
    <p>
      passengers: 
      30
    </p>
    <p>
      pilots: 
    </p>
    <p>
      const in credits: 
      150000
    </p>
  </div>
</li>
`;
